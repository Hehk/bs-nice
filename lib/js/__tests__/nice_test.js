// Generated by BUCKLESCRIPT VERSION 2.1.0, PLEASE EDIT WITH CARE
'use strict';

var Jest  = require("bs-jest/lib/js/src/jest.js");
var Nice  = require("../src/nice.js");
var Block = require("bs-platform/lib/js/block.js");

describe("Nice", (function () {
        return Jest.test("toBe", (function () {
                      var cls = Nice.css(/* :: */[
                            /* Display */Block.__(0, [/* Flex */5]),
                            /* [] */0
                          ]);
                      Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Nice.rule_cache));
                      return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](cls));
                    }));
      }));

/*  Not a pure module */
