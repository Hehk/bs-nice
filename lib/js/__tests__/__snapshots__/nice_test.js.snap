// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nice css 1`] = `
Array [
  "css-1lqfuW",
  Array [
    Array [
      ".css-1lqfuW",
      Array [
        ".css-1lqfuW{display:flex;width:23px}",
        0,
      ],
    ],
    0,
  ],
]
`;

exports[`Nice global 1`] = `
Array [
  Array [
    ".raw-24eOUU",
    Array [
      "html, body{padding:20px}",
      0,
    ],
  ],
  0,
]
`;

exports[`Nice nesting 1`] = `
Array [
  "css-1cUkzM",
  Array [
    Array [
      ".css-1cUkzM",
      Array [
        ".css-1cUkzM{display:flex;color:red}",
        Array [
          "@media screen{.css-1cUkzM{display:block;color:green}}",
          Array [
            "@media screen{.css-1cUkzM:hover{color:orange;font-weight:bold;overflow:hidden}}",
            Array [
              "@media screen{@supports (display: flex){.css-1cUkzM:hover{color:blue}}}",
              Array [
                "@media screen{.css-1cUkzM:hover:active{color:red;display:block}}",
                Array [
                  "@media screen{.ie6 .css-1cUkzM:hover:active{color:blue}}",
                  0,
                ],
              ],
            ],
          ],
        ],
      ],
    ],
    0,
  ],
]
`;
